src/mgr.sol

```diff
@@ -186,128 +166,140 @@ contract TinlakeManager {
         emit File(what, data);
-        if (what == "vow") vow = data;
-        else if (what == "daiJoin") daiJoin = JoinLike(data);
-        else if (what == "end")  end = EndLike(data);
-        else revert("Vat/file-unrecognized-param");
+        if (what == "urn") {
+            urn = MIP21UrnLike(data);
+            dai.approve(data, uint256(-1));
+        }
+        else if (what == "liq") {
+            liq = MIP21LiquidationLike(data);
+        }
+        else if (what == "owner") {
+            owner = data;
+        }
+        else revert("TinlakeMgr/file-unknown-param");
     }
 
     // --- Liquidation ---
-    function tell() public auth {
-        require(safe, "TinlakeManager/not-safe");
-        (uint256 ink, ) = vat.urns(ilk, address(this));
+    // triggers a soft liquidation of the DROP collateral
+    // a redeemOrder is submitted to receive DAI back
+    function tell() public {
+        require(safe, "TinlakeMgr/not-safe");
+        bytes32 ilk = GemJoinLike(urn.gemJoin()).ilk();
+
+        (,,, uint48 toc) = liq.ilks(ilk);
+        require(toc != 0, "TinlakeMgr/not-liquidated");
+
+        (, uint256 art) = vat.urns(ilk, address(urn));
+        (, uint256 rate, , ,) = vat.ilks(ilk);
+        tab = mul(art, rate);
+
+        uint256 ink = gem.balanceOf(address(this));
         safe = false;
         gem.approve(tranche, ink);
         pool.redeemOrder(ink);
         emit Tell(ink);
     }
 
+    // triggers the payout of a DROP redemption
+    // method can be called multiple times after the liquidation until all
+    // DROP tokens are redeemed
     function unwind(uint256 endEpoch) public {
-        require(!safe && glad && live, "TinlakeManager/not-soft-liquidation");
-        (uint256 redeemed, , ,uint256 remainingDrop) = pool.disburse(endEpoch);
-        (uint256 ink, uint256 art) = vat.urns(ilk, address(this));
-        uint256 dropReturned = sub(ink, remainingDrop);
-        require(dropReturned <= 2 ** 255, "TinlakeManager/overflow");
+        require(!safe && glad && live, "TinlakeMgr/not-soft-liquidation");
+
+        (uint256 redeemed, , ,) = pool.disburse(endEpoch);
+        bytes32 ilk = GemJoinLike(urn.gemJoin()).ilk();
 
+        (, uint256 art) = vat.urns(ilk, address(urn));
         (, uint256 rate, , ,) = vat.ilks(ilk);
-        uint256 cdptab = mul(art, rate);
-        uint256 payBack = min(redeemed, divup(cdptab, RAY));
-
-        daiJoin.join(address(this), payBack);
-        // Repay dai debt up to the full amount
-        // and exit the gems used up
-        uint256 dart = mul(RAY, payBack) / rate;
-        require(dart <= 2 ** 255, "TinlakeManager/overflow");
-        vat.frob(ilk, address(this), address(this), address(this),
-                 0, -int256(dart));
-        vat.grab(ilk, address(this), address(this), address(this),
-                 -int256(dropReturned), 0);
-        vat.slip(ilk, address(this), -int256(dropReturned));
+        tab = mul(art, rate);
```

Comment:
I think the actual update of  `tab` should happen after the `urn.wipe`. It would currently not include the `disburse redeemedDAI`. If I understand correctly